eland/filter.py

```diff
@@ -174,3 +174,28 @@ class QueryFilter(BooleanFilter):
     def __init__(self, query: Dict[str, Any]) -> None:
         super().__init__()
         self._filter = query
+
+
+class MatchAllFilter(QueryFilter):
+    def __init__(self) -> None:
+        super().__init__({"match_all": {}})
+
+
+class RandomScoreFilter:
+    def __init__(self, query, random_state):
+
+        q = MatchAllFilter() if query.empty() else query
+
+        seed = {}
+        if random_state is not None:
+            seed = {"seed": random_state, "field": "_seq_no"}
+
+        self._score = {"function_score": {"query": q.build(), "random_score": seed}}
```

Comment:
Let's instead use `super().__init__({"function_score": ...}) so that we can rely on code already using `_filter`.